mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/MantisJobStore.java

```diff
@@ -393,33 +393,25 @@ private class ArchivedJobsMetadataCache {
 
         IMantisJobMetadata getJob(String jobId) {
             try {
-                return cache.get(jobId, () -> loadArchivedJob(jobId));
-            } catch (ExecutionException e) {
+                final Optional<IMantisJobMetadata> jobMetadata = cache.get(jobId, () -> loadArchivedJobImpl(jobId));
+                return jobMetadata.orElse(null);
```

Comment:
Isn't this problematic? We should not cache "no value" as when there is a value available for the same key in the future, we will return "no value" as it was cached.

Also, feel that this is not addressing the core of the issue which is why are we missing the value in the first place and if the value can sometimes be missing, shouldn't the API take that into account when calling this?